### PR TITLE
Added some test cases, fixed bugs according the tests

### PR DESCRIPTION
**Pull request by stankevichevg**

### CODE REVIEW

File: java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/AnnotationExtensions.kt [hunk 1]
```diff
@@ -168,16 +174,18 @@ internal val Explode.model: Boolean? get() = when (this) {
 
 internal val String.model: String? get() = if (isNotBlank()) this else null
 
-internal val Array<String>.model: List<String>? get() =
+internal val Array<String>.model: List<String> get() =
```
Comment (ihostage) [inline java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/AnnotationExtensions.kt]:
```suggestion
internal val Array<String>.model: List<String> get() = filter { it.isNotBlank() }
```

File: java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/ServiceReader.kt [hunk 2]
```diff
@@ -33,9 +34,11 @@ import io.swagger.v3.oas.models.security.SecurityRequirement as ModelSecurityReq
 import io.swagger.v3.oas.models.servers.Server as ModelServer
 import io.swagger.v3.oas.models.tags.Tag as ModelTag
 
-internal class ServiceReader {
+class ServiceReader constructor(initialSpec: OpenAPI) {
```
Comment (ihostage) [inline java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/ServiceReader.kt]:
```suggestion
class ServiceReader(private val spec: OpenAPI = OpenAPI()) {
```

File: java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import io.swagger.v3.oas.annotations.media.Schema;
+import org.jetbrains.annotations.NotNull;
+import org.taymyr.lagom.openapi.AbstractOpenAPIService;
+import org.taymyr.lagom.openapi.OpenAPIService;
+
+import static com.lightbend.lagom.javadsl.api.Service.named;
+import static com.lightbend.lagom.javadsl.api.Service.restCall;
+
+/**
+ * @author Evgeny Stankevich {@literal <stankevich.evg@gmail.com>}.
```
Comment (ihostage) [inline java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java]: remove

File: java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/User.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
+/**
```
Comment (ihostage) [inline java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/User.java]: remove

File: java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java (hunk 3 above)
Comment (stankevichevg) [reply java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java]: Oops )

File: java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/AnnotationExtensions.kt (hunk 1 above)
Comment (stankevichevg) [reply java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/AnnotationExtensions.kt]: done

File: java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/ServiceReader.kt (hunk 2 above)
Comment (stankevichevg) [reply java/impl/src/main/kotlin/org/taymyr/lagom/openapi/internal/ServiceReader.kt]: done

File: java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java (hunk 3 above)
Comment (stankevichevg) [reply java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/ParameterService.java]: done

File: java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/User.java (hunk 4 above)
Comment (stankevichevg) [reply java/impl/src/test/java/org/taymyr/lagom/openapi/autogenerate/cases/User.java]: done

**Comment by ihostage** [comment]
LGTM

**Pull request merged**